toothpick-compiler/src/test/java/toothpick/compiler/factory/FactoryRegistryTest.java

```diff
@@ -24,14 +24,26 @@
         "package toothpick;", //
         "", //
         "import java.lang.Class;", //
+        "import java.lang.String;", //
         "import toothpick.registries.factory.AbstractFactoryRegistry;", //
         "", //
         "public final class FactoryRegistry extends AbstractFactoryRegistry {", //
         "  public FactoryRegistry() {", //
         "  }", //
         "", //
         "  public <T> Factory<T> getFactory(Class<T> clazz) {", //
-        "    switch(clazz.getName().replace('$','.')) {", //
+        "    String className = clazz.getName().replace('$','.');", //
+        "    int bucket = (className.hashCode() & 0);", //
```

Comment:
& 0 is fine for this case:

just 1 bucket --> all classes go to the same bucket